hypershift-operator/controllers/platform/aws/controller_test.go

```diff
@@ -157,3 +174,97 @@ func TestReconcileAWSEndpointServiceStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteAWSEndpointService(t *testing.T) {
+	tests := []struct {
+		name      string
+		ec2Client ec2iface.EC2API
+		expected  bool
+		expectErr bool
+	}{
+		{
+			name: "successful deletion",
+			ec2Client: &fakeEC2Client{
+				deleteOut: &ec2.DeleteVpcEndpointServiceConfigurationsOutput{
+					Unsuccessful: []*ec2.UnsuccessfulItem{},
+				},
+			},
+			expected:  true,
+			expectErr: false,
+		},
+		{
+			name: "endpoint service no longer exists",
+			ec2Client: &fakeEC2Client{
+				deleteOut: &ec2.DeleteVpcEndpointServiceConfigurationsOutput{
+					Unsuccessful: []*ec2.UnsuccessfulItem{
+						{
+							Error: &ec2.UnsuccessfulItemError{
+								Code:    aws.String("InvalidVpcEndpointService.NotFound"),
+								Message: aws.String("The VpcEndpointService Id 'vpce-svc-id' does not exist"),
+							},
+							ResourceId: aws.String("vpce-svc-id"),
+						},
+					},
+				},
+			},
+			expected:  true,
+			expectErr: false,
+		},
+		{
+			name: "existing connections",
+			ec2Client: &fakeEC2Client{
+				deleteOut: &ec2.DeleteVpcEndpointServiceConfigurationsOutput{
+					Unsuccessful: []*ec2.UnsuccessfulItem{
+						{
+							Error: &ec2.UnsuccessfulItemError{
+								Code:    aws.String("ExistingVpcEndpointConnections"),
+								Message: aws.String("Service has existing active VPC Endpoint connections!"),
+							},
+							ResourceId: aws.String("vpce-svc-id"),
+						},
+					},
+				},
+				describeOut: &ec2.DescribeVpcEndpointConnectionsOutput{
+					VpcEndpointConnections: []*ec2.VpcEndpointConnection{
+						{
+							VpcEndpointId:    aws.String("vpce-id"),
+							VpcEndpointState: aws.String("available"),
+						},
+					},
+				},
+			},
+			expected:  false,
```

Comment:
Just to clarify, Is this expectation because of the RejectVpcEndpointConnectionsWithContext call would fail in the test?

Reply:
Ah `RejectVpcEndpointConnectionsWithContext` would still succeed within the test, it's more that rejection doesn't happen instantaneously - when an endpoint has active connections we just make this AWS API call, but return false no matter what https://github.com/openshift/hypershift/pull/2700/files/e08f40b84c24498e9787771f1bea6203e429ac0d#diff-703bbd22baf7c689e315f5c561efed7afea4f14d0ade48d66af784206a0bc421R563-R574 with the intention of returning true in a future reconcile loop when `DescribeVpcEndpointConnections` finds 0 existing connections